mysys/mulalloc.c

```diff
@@ -90,11 +93,15 @@ void *my_multi_malloc_large(PSI_memory_key key, myf myFlags, ...)
     length=va_arg(args,ulonglong);
     tot_length+=ALIGN_SIZE(length);
   }
+  ret_total_length= va_arg(args, size_t *);
+  *ret_total_length= tot_length + MY_MEMORY_HEADER_SIZE;
   va_end(args);
 
-  if (!(start=(char *) my_malloc(key, (size_t) tot_length, myFlags)))
+  if (!(start=(char *) my_large_malloc(ret_total_length,
+                                       myFlags)))
     DBUG_RETURN(0); /* purecov: inspected */
 
+  start= my_psi_key_init(key, start, tot_length, myFlags);
```

Comment:
Wouldn't it be better to have my_large_malloc() call my_psi_key_init()?
This would benefit all callers of my_large_malloc() !